app/gui2/src/util/ast/abstract.ts

```diff
@@ -165,7 +165,31 @@ export function parseQualifiedName(ast: Ast): QualifiedName | null {
   return idents && normalizeQualifiedName(qnFromSegments(idents))
 }
 
-/* Substitute `pattern` inside `expression` with `to`.
+/** Substitute `pattern` inside `expression` with `to`.
+ * Will only replace the first item in the property acccess chain. */
+export function substituteIdentifier(
+  module: MutableModule,
+  expression: Ast,
+  pattern: IdentifierOrOperatorIdentifier,
+  to: IdentifierOrOperatorIdentifier,
+) {
+  const expr = module.getVersion(expression) ?? expression
+  if (expr instanceof Ident && expr.code() === pattern) {
+    expr.updateValue(() => Ast.parse(to, module))
+  } else if (expr instanceof PropertyAccess && expr.lhs instanceof Ident) {
```

Comment:
This exception is not general enough--test suggested above would fail.

Condition should be:
```suggestion
  } else if (expr instanceof PropertyAccess) {
```

Then recurse on `lhs` if `expr.lhs != null`.

Reply:
Applied